olive/model/config/io_config.py

```diff
@@ -80,6 +89,78 @@ def check_string_to_int_dim_params(cls, v):
                 raise ValueError(f"Invalid string_to_int_dim_params: {dim_param}. Must be castable to int.") from None
         return v
 
+    def get_seq_len(self):
+        if not self.input_shapes or not self.input_names:
+            return 0
+        for idx, name in enumerate(self.input_names):
+            if name == "input_ids":
+                return self.input_shapes[idx][1]  # pylint: disable=E1136
+        return 0
+
+    def get_past_seq_len(self):
+        if not self.input_shapes or not self.input_names:
+            return 0
+        attention_mask_len, seq_len = 0, 0
+        for idx, name in enumerate(self.input_names):
+            if name == "attention_mask":
+                attention_mask_len = self.input_shapes[idx][1]  # pylint: disable=E1136
+            if name == "input_ids":
+                seq_len = self.input_shapes[idx][1]  # pylint: disable=E1136
+        return attention_mask_len - seq_len
+
+    def get_batch_size(self):
+        if not self.input_shapes or not self.input_names:
+            return 1
+        for idx, name in enumerate(self.input_names):
+            if name == "input_ids":
+                return self.input_shapes[idx][0]  # pylint: disable=E1136
+        # if no input_ids, return 1
+        return 1
+
+
+def complete_kv_cache_with_model_attributes(kv_cache_config, model_attributes):
```

Comment:
>kv_cache_config [](http://example.com/codeflow?start=44&length=15)

for example, we could pass the num_heads names as string or int value here. If num_heads is string, we could search from model_attributes before constructing the KVCacheConfig instance at line 123